packages/core-components/src/layout/SignInPage/UserIdentity.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  IdentityApi,
+  ProfileInfo,
+  ProfileInfoApi,
+  BackstageUserIdentity,
+  BackstageIdentityApi,
+  SessionApi,
+  SignInResult,
+} from '@backstage/core-plugin-api';
+
+import { GuestUserIdentity } from './GuestUserIdentity';
+import { LegacyUserIdentity } from './LegacyUserIdentity';
+
+export class UserIdentity implements IdentityApi {
+  static createGuest() {
+    return new GuestUserIdentity();
+  }
+
+  static fromLegacy({ result }: { result: SignInResult }) {
```

Comment:
I think it's fine to make this `.fromLegacy(result)`, no need for the extra wrapping since it's deprecated anyway